src/tracker/index.ts

```diff
@@ -0,0 +1,97 @@
+import express from 'express'
+import { Tor } from '../torManager'
+import {ZBAY_DIR_PATH} from '../constants'
+import * as path from 'path'
+import * as os from 'os'
+import fs from 'fs'
+import multiaddr from 'multiaddr'
+
+export class Tracker {
+  private _app: express.Application
+  private _peers: Set<string>
+  private _port: number
+  private _controlPort: number
+
+  constructor(port?: number, controlPort?: number) {
+    this._app = express()
+    this._peers = new Set()
+    this._port = port || 7788
+    this._controlPort = controlPort || 9051
+  }
+
+  private async initTor() {
+    const torPath = `${process.cwd()}/tor/tor`
+    const pathDevLib = path.join.apply(null, [process.cwd(), 'tor'])
+    if(!fs.existsSync(ZBAY_DIR_PATH)) {
+      fs.mkdirSync(ZBAY_DIR_PATH)
+    }
+    const tor = new Tor({
+      torPath,
+      appDataPath: ZBAY_DIR_PATH,
+      controlPort: this._controlPort,
+      options: {
+        env: {
+          LD_LIBRARY_PATH: pathDevLib,
+          HOME: os.homedir()
+        },
+        detached: true
+      }
+    })
+    await tor.init()
+    return await tor.addOnion({ virtPort: this._port, targetPort: this._port, privKey: process.env.HIDDEN_SERVICE_SECRET })
```

Comment:
preferably you would also get `privKey` i constructor. This way this class is not making assumptions about how secret is passed in